tsl/src/continuous_aggs/Readme.md

```diff
@@ -1,144 +1,115 @@
+manually or automated by a policy that runs in the background. Unlike
+a regular materialized view, a continuous aggregate doesn't require
+complete re-materialization on every refresh. Instead, it is possible
+to refresh a subset of the continuous aggregate at relatively low
+cost, thus enabling continuous aggregation as new data is written or
+old data is updated and/or backfilled.
+
+To enable continuous aggregation, a continuous aggregate stores
+partial aggregations for every time bucket in an internal
+hypertable. The advantage of this configuration is that each time
+bucket can be recomputed individually, without requiring updates to
+other buckets, and buckets can be combined to form more granular
+aggregates (e.g., hourly buckets can be combined to form daily
+buckets). Finalization of the partial buckets happens automatically at
+query time. Although such finalization gives slightly higher querying
+times, it is offset by more efficient refreshes that only recompute
+the buckets that have been "invalidated" by changes in the raw data.
+
+A continuous aggregate policy automates the refreshing, allowing the
+aggregate to stay up-to-date without manual intervention. A policy can
+be configured to only refresh the most recent data (e.g., just the
+last hour's worth of data) or ensure that the continuous aggregate is
+always up-to-date with the underlying source data. Policies that focus
+on recent data allow older parts of the continuous aggregate to stay
+the same or be governed by manual refreshes.
+
+## Bookkeeping and Internal State ##
+
+TimescaleDB does bookkeeping for each continuous aggregate to know
+which buckets of the aggregates require refreshing. Whenever a
+modification happens to the source data, an invalidation for the
+modified region is written to an invalidation log. However,
+invalidations are not written after the *invalidation threshold*,
+which tracks the latest bucket materialized thus far. This threshold
+allows write amplification to be kept to a minimum by, e.g, never
+refreshing the most recent time bucket that is still being "filled".
+
+Thus, to store, maintain, and query aggregations, continuous
+aggregates consist of the following objects:
+
+1. A user view, which queries and finalizes the aggregations and is
+   also the object that users interact with.
+2. A partial view, which is used to materialize new data.
+3. A direct view, which holds the original query that users specified.
+4. An internal materialization hypertable, containing the materialized
+   data as partial aggregates for each time bucket.
+5. An invalidation threshold, which tracks how "far" the
+   materialization has reached and below which invalidations must be
+   logged.
+6. A trigger on the source hypertable that invalidates regions of data
+   in every transaction, based on INSERT, UPDATE, and DELETE
+   statements. The trigger writes its invalidations to the hypertable
+   invalidation log.
```

Comment:
It doesn't actually invalidate the regions, it just writes an invalidation record to the log. IMHO, it is better to be specific here.
```suggestion
6. A trigger on the source hypertable that writes invalidations of data
   for every transaction, based on INSERT, UPDATE, and DELETE
   statements. The trigger writes its invalidations to the hypertable
   invalidation log.
```